From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 18 May 2019 12:22:17 +0200
Subject: [PATCH] {BUG in 5.2-rc1} drivers/gpu/drm/i915/gvt/gtt: initialize ret

clang warns:

    drivers/gpu/drm/i915/gvt/gtt.c:1079:7: error: variable 'ret' is used
    uninitialized whenever 'if' condition is true
          [-Werror,-Wsometimes-uninitialized]
                    if (!gtt_type_is_pt(type))
                        ^~~~~~~~~~~~~~~~~~~~~
    drivers/gpu/drm/i915/gvt/gtt.c:1106:17: note: uninitialized use
    occurs here
            return ERR_PTR(ret);
                           ^~~
    drivers/gpu/drm/i915/gvt/gtt.c:1079:3: note: remove the 'if' if its
    condition is always false
                    if (!gtt_type_is_pt(type))
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/gpu/drm/i915/gvt/gtt.c:1054:9: note: initialize the variable
    'ret' to silence this warning
            int ret;
                   ^
                    = 0
---
 drivers/gpu/drm/i915/gvt/gtt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/gvt/gtt.c b/drivers/gpu/drm/i915/gvt/gtt.c
index 210016192ce7..1a83ff7926be 100644
--- a/drivers/gpu/drm/i915/gvt/gtt.c
+++ b/drivers/gpu/drm/i915/gvt/gtt.c
@@ -1072,7 +1072,7 @@ static struct intel_vgpu_ppgtt_spt *ppgtt_populate_spt_by_guest_entry(
 	struct intel_gvt_gtt_pte_ops *ops = vgpu->gvt->gtt.pte_ops;
 	struct intel_vgpu_ppgtt_spt *spt = NULL;
 	bool ips = false;
-	int ret;
+	int ret = 0 /* BUG */;
 
 	GEM_BUG_ON(!gtt_type_is_pt(get_next_pt_type(we->type)));
 
-- 
